mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/mapper/MapperDaoFactoryMethodGenerator.java

```diff
@@ -176,33 +210,64 @@ public MapperDaoFactoryMethodGenerator(
 
   private String validateKeyspaceOrTableParameter(
       VariableElement candidate, String previous, Class<?> annotation, ProcessorContext context) {
-    if (previous != null) {
+    if (!isSingleAnnotation(candidate, previous, annotation, context)) {
+      return null;
+    }
+    TypeMirror type = candidate.asType();
+    if (!context.getClassUtils().isSame(type, String.class)
+        && !context.getClassUtils().isSame(type, CqlIdentifier.class)) {
       context
           .getMessager()
           .error(
               candidate,
               processedType,
-              "Invalid parameter annotations: "
-                  + "only one %s method parameter can be annotated with @%s",
+              "Invalid parameter type: @%s-annotated parameter of %s methods must be of type %s or %s",
+              annotation.getSimpleName(),
               DaoFactory.class.getSimpleName(),
-              annotation.getSimpleName());
+              String.class.getSimpleName(),
+              CqlIdentifier.class.getSimpleName());
+      return null;
+    }
+    return candidate.getSimpleName().toString();
+  }
+
+  private String validateExecutionProfile(
+      VariableElement candidate, String previous, Class<?> annotation, ProcessorContext context) {
```

Comment:
IDEA warning: Actual value of parameter 'annotation' is always 'DaoProfile.class'